staging/src/k8s.io/apiserver/pkg/server/config_selfclient_test.go

```diff
@@ -43,8 +51,17 @@ func TestLoopbackHostPort(t *testing.T) {
 	if port != "443" {
 		t.Fatalf("expected 443 as port, got %q", port)
 	}
+}
+func TestLoopbackHostPortIPv6(t *testing.T) {
+	ipv6, _, err := isIPv6LoopbackSupported()
+	if err != nil {
+		t.Fatalf("fail to enumerate network interface, %s", err)
+	}
+	if !ipv6 {
+		t.Skipf("no ipv6 loopback interface")
```

Comment:
Shouldn't this still be a failure? You're skipping based on a property of the machine the test is running on. I agree it may be a failure that a dev on a crusty old system can ignore, but it's not a failure we'd be OK with our continuous testing ignoring--if our continuous testing doesn't exercise this case I want a red box, not a green-box-with-an-S.

Reply:
> Shouldn't this still be a failure? You're skipping based on a property of the machine the test is running on. I agree it may be a failure that a dev on a crusty old system can ignore, but it's not a failure we'd be OK with our continuous testing ignoring--if our continuous testing doesn't exercise this case I want a red box, not a green-box-with-an-S.

i hear you @lavalamp, although i think that is a different issue, which is to monitor skipping tests  and make sure skipping are expected (e.x provider specific, or machine type specific), we did this in internal testing system, i believe upstream kubernetes need something similar.

although i am open to the option: fail the test with very explicit message considering upstream kubernetes has not established skipping testing triage process (e.x use Yellow-box for skipping tests).

further, running ci on ipv6 only node seems possible in the future.